WordPress/src/main/java/org/wordpress/android/ui/mysite/SiteStoriesHandler.kt

```diff
@@ -0,0 +1,105 @@
+    private val storiesMediaPickerResultHandler: StoriesMediaPickerResultHandler
+) {
+    private val _onSnackbar = MutableLiveData<Event<SnackbarMessageHolder>>()
+    val onSnackbar = _onSnackbar as LiveData<Event<SnackbarMessageHolder>>
+    private val _onNavigation = MutableLiveData<Event<SiteNavigationAction>>()
+    val onNavigation = merge(_onNavigation, storiesMediaPickerResultHandler.onNavigation)
+
+    init {
+        eventBusWrapper.register(this)
+    }
+
+    fun clear() {
+        eventBusWrapper.unregister(this)
+    }
+
+    @Subscribe(sticky = true, threadMode = ThreadMode.MAIN)
+    fun onEventMainThread(event: StorySaveResult) {
+        EventBus.getDefault().removeStickyEvent(event)
+        if (!event.isSuccess()) {
+            // note: no tracking added here as we'll perform tracking in StoryMediaSaveUploadBridge
+            val errorText = String.format(
+                    resourceProvider.getString(R.string.story_saving_snackbar_finished_with_error),
+                    StoryRepository.getStoryAtIndex(event.storyIndex).title
+            )
+            val snackbarMessage = FrameSaveNotifier.buildSnackbarErrorMessage(
+                    contextProvider.getContext(),
+                    StorySaveEvents.allErrorsInResult(event.frameSaveResult).size,
+                    errorText
+            )
+
+            _onSnackbar.postValue(
+                    Event(
+                            SnackbarMessageHolder(
+                                    UiStringText(snackbarMessage),
+                                    UiStringRes(R.string.story_saving_failed_quick_action_manage)
+                            ) {
+                                val selectedSite = selectedSiteRepository.getSelectedSite()
+                                        ?: return@SnackbarMessageHolder
+                                _onNavigation.postValue(Event(OpenStories(selectedSite, event)))
+                                // TODO WPSTORIES add TRACKS: the putExtra described here below for NOTIFICATION_TYPE
+                                // is meant to be used for tracking purposes. Use it!
+                                // TODO add NotificationType.MEDIA_SAVE_ERROR param later when integrating with WPAndroid
+                                //        val notificationType = NotificationType.MEDIA_SAVE_ERROR
+                                //        notificationIntent.putExtra(ARG_NOTIFICATION_TYPE, notificationType)
+                                storiesTrackerHelper.trackStorySaveResultEvent(
+                                        event,
+                                        STORY_SAVE_ERROR_SNACKBAR_MANAGE_TAPPED
+                                )
+                            }
+                    )
+            )
+        }
+    }
+
+    @Subscribe(sticky = true, threadMode = ThreadMode.MAIN)
+    fun onStorySaveStart(event: StorySaveProcessStart) {
+        EventBus.getDefault().removeStickyEvent(event)
```

Comment:
Same thing as above.
```suggestion
        eventBusWrapper.removeStickyEvent(event)
```